File: ERP_2021_2022_Grupo_1/ERP_2021_2022_Grupo_1_Entities/clsOrder.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace ERP_2021_2022_Grupo_1_Entities
{
    public class clsOrder
    {
        #region attributes
        private int orderId;
        private double total;
        private DateTime orderDate;
        private DateTime limitOrderDate;
        private string notes;
        private int supplierId;
        #endregion
        #region constructor
        public clsOrder(int orderId, double total, DateTime orderDate, DateTime limitOrderDate, string notes, int supplierId)
        {
            this.orderId = orderId;
            this.total = total;
            this.orderDate = orderDate;
            this.limitOrderDate = limitOrderDate;
            this.notes = notes;
            this.supplierId = supplierId;
        }
        public clsOrder(double total, DateTime orderDate, DateTime limitOrderDate, string notes, int supplierId)
        {
            this.total = total;
            this.orderDate = orderDate;
            this.limitOrderDate = limitOrderDate;
            this.notes = notes;
            this.supplierId = supplierId;
        }
        public clsOrder()
        {
            this.orderId = 0;
            this.total = 0;
            this.orderDate = DateTime.Now;
            this.limitOrderDate = DateTime.Now;
            this.notes = "notes";
            this.supplierId = 0;
        }
        #endregion
        #region public methods
        public int OrderId { get => orderId; set => orderId = value; }
        public double Total { get => total; set => total = value; }
        public DateTime OrderDate { get => orderDate; set => orderDate = value; }
        public DateTime LimitOrderDate { get => limitOrderDate; set => limitOrderDate = value; }
        public string Notes { get => notes; set => notes = value; }
        public int SupplierId { get => supplierId; set => supplierId = value; }
        #endregion
    }
}

File: ERP_2022_Grupo1/src/app/model/cls-order.ts
class ClsOrder {
    //Attributes
    private _orderId: number;
    private _total: number;
    private _orderDate: Date;
    private _limitOrderDate: Date;
    private _notes: String;
    private _supplierId: number;

    //Constructor
    constructor(orderId: number,total: number,orderDate: Date,limitOrderDate: Date,notes: String,supplierId: number){
        this._orderId = orderId;
        this._total = total;
        this._orderDate = orderDate;
        this._limitOrderDate = limitOrderDate;
        this._notes = notes;
        this._supplierId = supplierId;
    }

    //Getters & Setters 
    get orderId() : number{
        return this._orderId;
    }
    set orderId(orderId: number){
        this._orderId = orderId;
    }
    get total(): number{
        return this._total;
    }
    set total(total: number){
        this._total = total;
    }
    get orderDate() : Date{
        return this._orderDate;
    }
    set orderDate(orderDate: Date){
        this._orderDate = orderDate;
    }
    get limitOrderDate() : Date{
        return this._limitOrderDate;
    }
    set limitOrderDate(limitOrderDate: Date){
        this._limitOrderDate = limitOrderDate;
    }
    get notes() : String{
        return this._notes;
    }
    set notes(notes: String){
        this._notes = notes;
    }
    get supplierId() : number{
        return this._supplierId;
    }
    set supplierId(supplierId: number){
        this._supplierId = supplierId;
    }
}
